packages/@lwc/rollup-plugin/src/index.ts

```diff
@@ -0,0 +1,147 @@
+/*
```

Comment:
This is the meat of the PR. This file has been completely rewritten in TypeScript.